hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/CommittableWritableStates.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.state;
+
+import com.hedera.node.app.spi.workflows.HandleContext.SavepointStack;
+
+/**
+ * A {@link WritableStates} implementation at the "bottom" of a {@link SavepointStack}; i.e., an
+ * implementation that is not buffering changes for a wrapped delegate, but itself knows how to
+ * persist changes.
+ */
+public interface CommittableWritableStates {
```

Comment:
I'm skeptical of this interface. We've avoided adding it up to this point. Was it mostly to avoid the instanceof check?

Reply:
Yes, I thought it would be odd if the production code had an `instanceof MapWritableStates` here. 🤔